### PR TITLE
Add documentation for configuring the password hashing algorithm and its properties

### PR DESCRIPTION
**Pull request by dancristiancecoi**
### Description
Added documentation for configuring the password hashing algorithm and its properties

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/7669
### Version
2.16. 3.x 
### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dancristiancecoi** [comment]
Hey @scrawfor99 could I get this reviewed please so we have it ready for 2.16?

**Review by scrawfor99 — APPROVED** [review]

File: _install-and-configure/configuring-opensearch/security-settings.md [hunk 1]
```diff
@@ -122,6 +122,41 @@ The Security plugin supports the following expert-level settings:
...
+{: .warning}
+
+- `plugins.security.password.hashing.algorithm`: (Static): Specifies the password hashing algorithm to use.
+
+  Valid values are:
+  
+  - `BCrypt` (Default)
+  - `PBKDF2` 
+
+- `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with BCrypt. Valid values are from `4` to `31`, inclusive. Default is `12`.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/security-settings.md:135]: What do we mean by inclusive?

```suggestion
- `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with `BCrypt`. Valid values are a range from `4` to `31`, inclusive. Default is `12`.
```

File: _install-and-configure/configuring-opensearch/security-settings.md [hunk 2]
```diff
@@ -122,6 +122,41 @@ The Security plugin supports the following expert-level settings:
...
+
+- `plugins.security.password.hashing.bcrypt.minor` (Static): Specifies the minor version of the BCrypt algorithm to use for password hashing.
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/security-settings.md:137]:
```suggestion
- `plugins.security.password.hashing.bcrypt.minor` (Static): Specifies the minor version of the `BCrypt` algorithm to use for password hashing.
```

**Review by Naarcha-AWS — APPROVED** [review]
A couple of suggestions and a clarification. Otherwise, LGTM.

File: _install-and-configure/configuring-opensearch/security-settings.md (hunk 1 above)
Comment (dancristiancecoi) [reply _install-and-configure/configuring-opensearch/security-settings.md:135]: Just meant that the range includes 4 and 31 not just the numbers between them

Comment (dancristiancecoi) [reply _install-and-configure/configuring-opensearch/security-settings.md:135]: Thoughts on "Valid values range from `4` to `31`, inclusive." ?

Comment (kolchfa-aws) [reply _install-and-configure/configuring-opensearch/security-settings.md:135]: @dancristiancecoi Alternatively, feel free to use the math notation for ranges (`[4, 31]`).

Comment (Naarcha-AWS) [reply _install-and-configure/configuring-opensearch/security-settings.md:135]: Updated the suggestion accordingly.

Comment (dancristiancecoi) [reply _install-and-configure/configuring-opensearch/security-settings.md]: Thanks a lot! Just a tiny thing @Naarcha-AWS, was that space removed on purpose?

File: _install-and-configure/configuring-opensearch/security-settings.md [hunk 3]
```diff
@@ -122,6 +122,41 @@ The Security plugin supports the following expert-level settings:
...
+{: .warning}
+
+- `plugins.security.password.hashing.algorithm`: (Static): Specifies the password hashing algorithm to use.
+
+  Valid values are:
+  
+  - `BCrypt` (Default)
+  - `PBKDF2` 
+
+- `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with `BCrypt`. Valid values are from`4` to `31`, inclusive. Default is `12`.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/security-settings.md:135]:
```suggestion
- `plugins.security.password.hashing.bcrypt.rounds` (Static): Specifies the number of rounds to use for password hashing with `BCrypt`. Valid values are between `4` and `31`, inclusive. Default is `12`.
```

File: _install-and-configure/configuring-opensearch/security-settings.md [hunk 4]
```diff
@@ -122,6 +122,41 @@ The Security plugin supports the following expert-level settings:
...
+
+  Valid values are:
+
+  - `SHA1`
+  - `SHA224`
+  - `SHA256` (Default)
+  - `SHA384`
+  - `SHA512`
+
+- `plugins.security.password.hashing.pbkdf2.iterations` (Static): Specifies the number of times the pseudo-random function is applied to the password. Default is `600,000`.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/security-settings.md:155]:
```suggestion
- `plugins.security.password.hashing.pbkdf2.iterations` (Static): Specifies the number of times that the pseudo-random function is applied to the password. Default is `600,000`.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS @dancristiancecoi Just a couple of changes. Thanks!

**Pull request merged**